src/link_prober/LinkProber.cpp

```diff
@@ -313,25 +338,41 @@ void LinkProber::handleRecv(
                 mMuxPortConfig.getPortName() %
                 mMuxPortConfig.getBladeIpv4Address().to_string()
             );
-            if (memcmp(icmpPayload->un.uuid.data, IcmpPayload::getGuidData(),
-                       IcmpPayload::getGuid().size()) == 0) {
+            bool isMatch = (memcmp(icmpPayload->uuid, IcmpPayload::getGuidData(), sizeof(icmpPayload->uuid)) == 0);
+            if (isMatch)
+            {
                 MUXLOGTRACE(boost::format("%s: Matching Guid") % mMuxPortConfig.getPortName());
                 // echo reply for an echo request generated by this/active ToR
                 mRxSelfSeqNo = mTxSeqNo;
                 mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpSelfEvent());
-            } else {
-                // echo reply for an echo request generated by peer ToR
-                mRxPeerSeqNo = mTxSeqNo;
-                mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpPeerEvent());
-                if (ntohl(icmpPayload->command) == static_cast<uint32_t> (Command::COMMAND_SWITCH_ACTIVE)) {
-                    boost::asio::io_service::strand &strand = mLinkProberStateMachine.getStrand();
-                    boost::asio::io_service &ioService = strand.context();
-                    ioService.post(strand.wrap(boost::bind(
-                        static_cast<void (LinkProberStateMachine::*) (SwitchActiveRequestEvent&)>
-                            (&LinkProberStateMachine::processEvent),
-                        &mLinkProberStateMachine,
-                        LinkProberStateMachine::getSwitchActiveRequestEvent()
-                    )));
+            }
+
+            uint8_t tlvStartOffset = mPacketHeaderSize + sizeof(IcmpPayload);
+            bool stopProcessTlv = false;
+            while (tlvStartOffset + sizeof(Tlv) <= bytesTransferred) {
+                Tlv *currentTlv = reinterpret_cast<Tlv *> (mTxBuffer.data() + tlvStartOffset);
+                size_t currentTlvLength = ntohs(currentTlv->length);
+                switch (currentTlv->type) {
+                    case TlvType::TLV_COMMAND: {
+                        handleTlvCommandRecv(tlvStartOffset, !isMatch);
+                        break;
+                    }
+                    case TlvType::TLV_SENTINEL: {
+                        // sentinel TLV, stop processing
+                        stopProcessTlv = true;
+                        break;
+                    }
+                    default: {
+                        // unknonw TLV, try to skip
+                        if (currentTlvLength == 0) {
+                            stopProcessTlv = true;
+                            break;
+                        }
+                    }
+                    tlvStartOffset += sizeof(Tlv) + currentTlvLength;
+                }
+                if (stopProcessTlv) {
+                    break;
                 }
             }
```

Comment:
Add a debug event here if we didn't reach the end of the received bytes, but stopped by sentinel. state how many bytes missed.

Reply:
updated, thanks!